lib/provider/appium.ts

```diff
@@ -17,19 +17,11 @@ export class Appium implements ProviderInterface {
   proxy: string;
   requestUrl = 'http://registry.npmjs.org/appium';
 
-  constructor(providerConfig?: ProviderConfig) {
-    if (providerConfig) {
-      this.ignoreSSL = providerConfig.ignoreSSL;
-      if (providerConfig.outDir) {
-        this.outDir = providerConfig.outDir;
-      }
-      if (providerConfig.proxy) {
-        this.proxy = providerConfig.proxy;
-      }
-      if (providerConfig.requestUrl) {
-        this.requestUrl = providerConfig.requestUrl;
-      }
-    }
+  constructor(config?: ProviderConfig) {
+    this.ignoreSSL = setVar('ignoreSSL', this.ignoreSSL, config);

```

Comment:
Just a thought - you could also make Provider and abstract base class that implements setVar, then have all of these extend that. Then you wouldn't have to export setVar, it could live in the base Provider class.

Reply:
Per offline discussion, this would be a good idea. I guess we could have the class extend the class with the `setVar` method and implement the provider interface. This seems reasonable.